x-pack/plugins/security_solution/public/management/cypress/e2e/endpoint_list/endpoints_rbac_mocked_data.cy.ts

```diff
@@ -0,0 +1,201 @@
+              (fleetPrivilege === 'all' && integrationsPrivilege === 'all');
+
+            it(`should ${
+              shouldShowOnboardingSteps ? '' : ' NOT '
+            } show onboarding steps with fleet:${fleetPrivilege} and integrations:${integrationsPrivilege}`, () => {
+              loginWithCustomRole({
+                endpointPolicyManagementPrivilege,
+                fleetPrivilege,
+                integrationsPrivilege,
+              });
+
+              loadPage(APP_ENDPOINTS_PATH);
+
+              if (shouldShowOnboardingSteps) {
+                cy.getByTestSubj('emptyHostsTable', { timeout: 3000 }).should('exist');
+                cy.getByTestSubj('onboardingSteps', { timeout: 3000 }).should('exist');
+              } else {
+                // without correct privileges, fall back to empty policy table note showing that Fleet privilege is required
+                cy.getByTestSubj('emptyPolicyTable', { timeout: 3000 }).should('exist');
+                cy.getByTestSubj('onboardingStartButton', { timeout: 3000 }).should('not.exist');
+              }
+            });
+          }
+        }
+      });
+    }
+  });
+
+  describe('some hosts are enrolled', () => {
+    let endpointData: ReturnTypeFromChainable<typeof indexEndpointHosts>;
+
+    before(() => {
+      indexEndpointHosts({ count: 1 }).then((indexEndpoints) => {
+        endpointData = indexEndpoints;
+      });
+    });
+
+    after(() => {
+      if (endpointData) {
+        endpointData.cleanup();
+        // @ts-expect-error ignore setting to undefined
+        endpointData = undefined;
+      }
+    });
+
+    beforeEach(() => {
+      // if there is a request towards this API, it should return 200
+      cy.intercept(PACKAGE_POLICY_API_ROUTES.BULK_GET_PATTERN, (req) => {
+        req.on('response', (res) => {
+          expect(res.statusCode).to.equal(200);
+        });
+      });
+    });
+
+    for (const endpointPolicyManagementPrivilege of PRIVILEGES) {
+      describe(`endpoint policy management privilege is ${endpointPolicyManagementPrivilege}`, () => {
+        for (const fleetPrivilege of PRIVILEGES) {
+          for (const integrationsPrivilege of PRIVILEGES) {
+            const shouldProvidePolicyLink = endpointPolicyManagementPrivilege !== 'none';
+
+            it(`should show Endpoint list ${
+              shouldProvidePolicyLink ? 'with' : 'without'
+            } link to Endpoint Policy with fleet:${fleetPrivilege} and integrations:${integrationsPrivilege}`, () => {
+              loginWithCustomRole({
+                endpointPolicyManagementPrivilege,
+                fleetPrivilege,
+                integrationsPrivilege,
+              });
+
+              loadPage(APP_ENDPOINTS_PATH);
+
+              cy.getByTestSubj('policyNameCellLink', { timeout: 3000 }).should('exist');
+              cy.getByTestSubj('policyNameCellLink', { timeout: 3000 }).within(() => {
```

Comment:
Timeout seems low here (`3s`)... was this intentional?

Reply:
thanks for catching that! it was not intentional to keep them. i shortened the timeouts at the start of implementation, so the test used as an acceptance test capturing the bugs fails faster on my machine and on CI, but now we don't need these ✅ 
99a2c18a26129ad903532106c4d51c8f12356b54